nomad/vault.go

```diff
@@ -1231,13 +1257,21 @@ func (v *vaultClient) revokeDaemon() {
 			}
 
 			// Build the list of accessors that need to be revoked while pruning any TTL'd checks
-			revoking := make([]*structs.VaultAccessor, 0, len(v.revoking))
+			toRevoke := len(v.revoking)
+			if toRevoke > maxVaultRevokeBatchSize {
+				toRevoke = maxVaultRevokeBatchSize
+			}
+			revoking := make([]*structs.VaultAccessor, 0, toRevoke)
 			for va, ttl := range v.revoking {
 				if now.After(ttl) {
 					delete(v.revoking, va)
```

Comment:
In chatting with @notnoop we realized this likely leaks expired accessors in Nomad forever. They're expired on the Vault side but as far as I can tell we skip removing them from Nomad's statestore (via purgeFn below, which only operates on the non-expired batch).